### DISCUSSION TITLE
Maven manager does not parse environment variables of settings files

### DISCUSSION THREAD

**Discussion by braunsonm** [renovatebot/renovate#14457] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.95.0

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Given a maven project with a `settings.xml` with the following:

```xml
<settings>
  <profiles>
    <profile>
      <id>private-repo</id>
      <repositories>
        <repository>
          <id>private-repo</id>
          <name>Private Repo</name>
          <url>${env.REPO_URL}</url>
        </repository>
      </repositories>
    </profile>
  </profiles>
  <activeProfiles>
    <activeProfile>private-repo</activeProfile>
  </activeProfiles>
</settings>
```

The maven manager will detect this file but it will not properly expand the `REPO_URL` environment variable and thus will not be able to find private dependencies. I have set the environment variables like so in the `renovate.json`:

```json
"customEnvVariables": {
  "REPO_URL": "example.com"
}
```

You can sort of get around this by manually specifying the `registryUrls` but this will overwrite any other configuration and the default maven central URLs so this is not ideal.

Related to this PR: https://github.com/renovatebot/renovate/pull/13592

### Relevant debug logs

<details><summary>Logs</summary>

NOTE how registries is set to an empty array.

```
{"name":"renovate","hostname":"f63f57c73ff6","pid":9,"level":20,"logContext":"nt9N7hlHUZLmGlnOyIojJ","repository":"test/example","registries":[],"packageFile":"settings.xml","msg":"Found registryUrls in settings.xml","time":"2022-02-28T17:27:11.366Z","v":0}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
This is not a bug. It's working as intended. Please open a discussion first. Maybe we can allow this as a feature request.

**Reply by suneetk92** [reply]
I am facing the similar issue.
What is the intended behaviour here? Shouldn’t renovate first resolve the environment variables and then check for new versions?

**Comment by rarkins** [comment]
Env variable interpolation in managers is not a supported feature. It would require:
- `customEnvVariables` to be passed to each manager's "extract" function
- Each manager would need to implement this env interpolation separately, because the syntax across managers is not consistent

Alternatively if you wanted a smaller solution only for repository URLs, this could potentially be implemented using `registryAliases`

**Reply by suneetk92** [reply]
`maven` is not amongst the supported managers for `registryAliases`
<html><body>


supportedManagers | helm-requirements, helmv3, helmfile, gitlabci, dockerfile, docker-compose, kubernetes, ansible, droneci, woodpecker
-- | --



</body>
</html>

**Reply by rarkins** [reply]
Exactly, that's why I wrote "potentially be implemented". Would you like to submit a PR?

**Reply by suneetk92** [reply]
I will try to work on it over the weekends.
